QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,145 @@
+There is documentation for nearly all of the public API.
+
+### License [3.iii]
+
+The license for `foonathan_memory` is zlib, a summary statement is provided in each source file, and a full copy of the license is in the [LICENSE](https://raw.githubusercontent.com/foonathan/memory/master/LICENSE) file.
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `foonathan_memory`.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+Many features in `foonathan_memory` have corresponding tests which simulate typical usage, and they are located in the `test` directory.
+
+### Public API Testing [4.ii]
+
+Many parts of the public API have tests.
+
+### Coverage [4.iii]
+
+`foonathan_memory` does not track testing coverage.
+
+### Performance [4.iv]
+
+`foonathan_memory` does not conduct performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`foonathan_memory` does not conduct tests with linters on its public CI.
+
+## Dependencies [5]
+
+`foonathan_memory` does not have any dependencies outside of the C++ standard library.
+
+## Platform Support [6]
+
+`foonathan_memory` supports the following platforms and tests each one:
+- Ubuntu 16.04 for gcc 4.9 and 5.5
+- MacOS with XCode 9.4 with Apple LLVM version 9.1.0
+- Windows with Microsoft Visual Studio 2015, MSVC 18.0
+
+ARM 64 support is not directly evaluated by this package's CI process.
+
+Regular CI results can be seen here:
+* [Ubuntu and OS X](https://travis-ci.org/github/foonathan/memory).
+* [Windows](https://ci.appveyor.com/project/foonathan/memory/branch/master).
+
+## Vulnerability Disclosure Policy [7.i]
+
+`foonathan_memory` does not have a Vulnerability Disclosure Policy
+
+
+## Third-Party Package Quality Level Summary [8]
+
+As `foonathan_memory` is a third-party package in maintenance mode, it is unlikely that it will be able to accomplish the remaining steps to become a Quality Level 1 type of package.
+However, as part of the ROS 2 ecosystem it will be important that it can be validated to the equivalent quality level.
+
+Even though `foonathan_memory` by itself will not likely reach the equivalent level of quality as Quality Level 1, there are steps that can be taken by ROS contributors to ensure that its incorporation into ROS packages can provide the equivalent level of quality.
+ROS contributors will need to conduct coverage tests to identify the remaining API and features that are not currently covered by tests.
+If features and portions of the public API that are used in ROS dependencies are found to be untested, then the appropriate tests will be required.
+To solidify version stability, `foonathan_memory_vendor` shall declare a targeted version for each ROS 2 distribution release.
+If a newer version is desired for a previously released ROS 2 distribution, then it will need to be verified that the API does change between versions.
```

Comment:
I think this is a typo
```suggestion
If a newer version is desired for a previously released ROS 2 distribution, then it will need to be verified that the API does not change between versions.
```

Reply:
Fixed.